slatedb/src/sst_iter.rs

```diff
@@ -168,7 +164,13 @@ impl<'a> SstIterator<'a> {
         table_store: Arc<TableStore>,
         options: SstIteratorOptions,
     ) -> Result<Self, SlateDBError> {
-        let view = SstView::Owned(table, BytesRange::from(range));
+        let mut view_range = BytesRange::from(range);
+        if let Some(visible_range) = &table.visible_range {
+            view_range = view_range
+                .intersect(visible_range)
+                .expect("Provided range is outside of table's visible range");
```

Comment:
why are we panic'ing in this case? I think it's expected to hit this case if the client tries to do a scan outside the db's projected range - in which case we should just return an empty iterator.